ejb3/src/main/java/org/jboss/as/ejb3/remote/AssociationImpl.java

```diff
@@ -428,8 +428,7 @@ static Object invokeMethod(final ComponentView componentView, final Method metho
     }
 
     private static Object invokeWithIdentity(final ComponentView componentView, final InterceptorContext interceptorContext, final SecurityIdentity securityIdentity) throws Exception {
-        return securityIdentity == null ? componentView.invoke(interceptorContext) : securityIdentity.runAs((PrivilegedExceptionAction<Object>) () -> {
-            // TODO: replace this with identity.runAsFunctionEx() once it is available
+        return securityIdentity == null ? componentView.invoke(interceptorContext) : securityIdentity.runAsSupplierEx((ExceptionSupplier<Object, Exception>) () -> {
```

Comment:
Looking at this again, the cast is not needed.  Acutally it probably just could be: runAsFunctionEx(ComponentView::invoke, componentView, interceptorContext);